rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,152 @@
+
+**Potential issues from developer point of view:**
+
+- inconsistent file naming
+  - some files contain suffix (`.types`)
+  - some files have `.ts` some `.tsx` what's the difference ?
+  - some files are PascalCased, some camelCased
+- not explicitly communicating the intent
+  - what does `use` mean ?
+    - seasoned react dev will probably guess that it's a hook. Thing is, that this file might export more that 1 hook.
+
+## Detailed Design or Proposal
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+**Proposed rules applied on current state:**
+
+> see [Explanation](#Explanation) for more details about applied rules
+
+| Current File Name   | Proposed File name  | Symbol Name                                                                         | Remarks                                                                                                                                                                                                                                                                                                                   |
+| ------------------- | ------------------- | ----------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| index.tsx           | index.ts            |                                                                                     | No changes from current                                                                                                                                                                                                                                                                                                   |
+| Example.tsx         | example.ts          | `export const Example = () => {..}`                                                 | To enforce rendering logic only within `.view.` type file/s. TS will enforce this by default. Not extremely picky on this one, we can use `.tsx` as well to match extension with test/story - priority is to be consistent                                                                                                |
+| Example.test.tsx    | example.spec.tsx    | `describe->it`                                                                      | Integration test on JSDOM level (jest) for whole component                                                                                                                                                                                                                                                                |
```

Comment:
```suggestion
| Example.test.tsx    | Example.spec.tsx    | `describe->it`                                                                      | Integration test on JSDOM level (jest) for whole component                                                                                                                                                                                                                                                                |
```

If we will go with Pascal Case, this should be also in Pascal Case: https://github.com/microsoft/fluentui/pull/16870/files#r574385390